src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/ReplyToMessage.vue

```diff
@@ -45,26 +65,109 @@
 </template>
 
 <script lang="ts">
-import { defineComponent } from "vue";
+import type { AxiosProgressEvent } from "axios";
+import { v4 as uuidv4 } from "uuid";
+import { defineComponent, ref } from "vue";
 import { useRouter } from "vue-router";
 
+import { uploadFile } from "@/api/file";
 import { sendMessage } from "@/api/message";
 import ConfirmationDialog from "@/components/ConfirmationDialog.vue";
 import PageContainer from "@/components/PageContainer.vue";
+import UploadFileItem from "@/components/UploadFileItem.vue";
 import { useAlertStore } from "@/store/alert";
 import { useLoadingStore } from "@/store/loading";
 import { useMessageStore } from "@/store/message";
 import * as Rules from "@/utils/formRules";
 
 export default defineComponent({
   name: "ReplyToMessage",
-  components: { PageContainer, ConfirmationDialog },
+  components: { PageContainer, ConfirmationDialog, UploadFileItem },
   setup() {
     const messageStore = useMessageStore();
     const loadingStore = useLoadingStore();
     const alertStore = useAlertStore();
     const router = useRouter();
-    return { messageStore, loadingStore, alertStore, router };
+    const fileInput = ref<HTMLInputElement | null>(null);
+    const selectedFiles = ref<{ file: File; progress: number }[]>([]);
+    const formatFileTags = (file: File): string => {
+      const fileName = file.name;
+      const fileSize = (file.size / 1024).toFixed(2) + " KB"; // Convert size to KB
+      const fileFormat = file.name.split(".").pop(); // Get file extension
+
+      const tags = {
+        Name: fileName,
+        Size: fileSize,
+        Format: fileFormat,
+      };
+
+      return Object.entries(tags)
+        .map(([key, value]) => `${key}=${value}`)
+        .join(",");
+    };
+    const triggerFileInput = () => {
+      fileInput.value?.click();
+    };
+    const handleFileUpload = (event: Event) => {
+      const target = event.target as HTMLInputElement;
+      const files = target.files;
+      if (files) {
+        for (let i = 0; i < files.length; i++) {
+          const file = files[i];
+          if (file.size > 10 * 1024 * 1024) {
+            alertStore.setFailureAlert("File size exceeds the 10MB limit.");
+            continue; // Skip this file and continue with the next one
+          }
+          if (selectedFiles.value.some((f) => f.file.name === file.name)) {
+            alertStore.setFailureAlert("File with the same name already exists.");
+            continue; // Skip this file and continue with the next one
+          }
+          selectedFiles.value.push({ file, progress: 0 });
+          uploadFileWithProgress(file);
+        }
+      }
+    };
```

Comment:
we're missing validation here for the file type. ex. checking that it is .pdf

because I can upload a png

![image](https://github.com/bcgov/ECC-ECER/assets/74216496/1529fd93-e658-4517-b88e-97246fe1647d)

Reply:
This is not part of this story, we have ECER-2266 for validation of file size and file types. I just did some part of validation here.